corehq/apps/sms/models.py

```diff
@@ -582,52 +592,61 @@ def by_couch_id(cls, couch_id):
             return None
 
     @classmethod
-    def by_phone(cls, phone_number, include_pending=False):
-        result = cls._by_phone(apply_leniency(phone_number))
-        return cls._filter_pending(result, include_pending)
+    def get_two_way_number(cls, phone_number):
+        return cls._get_two_way_number(apply_leniency(phone_number))
 
     @classmethod
-    def by_suffix(cls, phone_number, include_pending=False):
-        """
-        Used to lookup a PhoneNumber, trying to exclude country code digits.
-        """
-        result = cls._by_suffix(apply_leniency(phone_number))
-        return cls._filter_pending(result, include_pending)
+    @quickcache(['phone_number'], timeout=60 * 60)
+    def _get_two_way_number(cls, phone_number):
+        try:
+            return cls.objects.get(phone_number=phone_number, is_two_way=True)
+        except cls.DoesNotExist:
+            return None
+
+    @classmethod
+    def get_number_pending_verification(cls, phone_number):
+        return cls._get_number_pending_verification(apply_leniency(phone_number))
 
     @classmethod
     @quickcache(['phone_number'], timeout=60 * 60)
-    def _by_phone(cls, phone_number):
+    def _get_number_pending_verification(cls, phone_number):
         try:
-            return cls.objects.get(phone_number=phone_number)
+            return cls.objects.get(
+                phone_number=phone_number,
+                verified=False,
+                pending_verification=True
+            )
         except cls.DoesNotExist:
             return None
 
     @classmethod
-    def _by_suffix(cls, phone_number):
-        # Decided not to cache this method since in order to clear the cache
-        # we'd have to clear using all suffixes of a number (which would involve
-        # up to ~10 cache clear calls on each save). Since this method is used so
-        # infrequently, it's better to not cache vs. clear so many keys on each
-        # save. Once all of our IVR gateways provide reliable caller id info,
-        # we can also remove this method.
+    def get_reserved_number(cls, phone_number):
```

Comment:
my thinking here was that both of those calls are cached so it'd be better/faster to lookup to the cache rather than the db